### PR TITLE
chore: remove App plugin

### PR DESCRIPTION
**Pull request by jcesarmobile**
Since on Android the App plugin was very coupled with the bridge I've created a new App class (that is not a plugin) to host that functionality and the App plugin will use it as an intermediary, i.e subscribe to its events.

### CODE REVIEW

File: android/capacitor/src/main/java/com/getcapacitor/App.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+     */
+    public interface AppRestoredListener {
+        void onAppRestored(PluginResult result);
+    }
+
+    @Nullable
+    private AppStatusChangeListener statusChangeListener;
+
+    private BackButtonListener backButtonListener;
+    private AppRestoredListener appRestoredListener;
```
Comment (dwieeb) [inline android/capacitor/src/main/java/com/getcapacitor/App.java:32]: Are these also `@Nullable`? or does it not matter?

File: android/capacitor/src/main/java/com/getcapacitor/App.java [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        this.appRestoredListener = listener;
+    }
+
+    protected void fireRestoredResult(PluginResult result) {
+        if (appRestoredListener != null) {
+            appRestoredListener.onAppRestored(result);
+        }
+    }
+
+    public void fireChange(boolean isActive) {
```
Comment (dwieeb) [inline android/capacitor/src/main/java/com/getcapacitor/App.java:69]: Would this be more descriptive?

```suggestion
    public void fireStatusChange(boolean isActive) {
```

**Review by dwieeb — APPROVED** [review]

**Comment by dwieeb** [comment]
Would it make sense to allow multiple listeners to these events, instead of just one?

**Comment by carlpoole** [comment]
> Would it make sense to allow multiple listeners to these events, instead of just one?

I wondered that too. I think it makes sense to allow for it

**Review by carlpoole — APPROVED** [review]

**Pull request merged**